class.jetpack.php

```diff
@@ -436,56 +436,68 @@ class_exists( 'Jetpack_Sitemap_Manager' )
 		delete_transient( self::$plugin_upgrade_lock_key );
 	}
 
-	static function update_active_modules( $modules ) {
-		$current_modules = Jetpack_Options::get_option( 'active_modules', array() );
-
-		$success = Jetpack_Options::update_option( 'active_modules', array_unique( $modules ) );
-
-		if ( is_array( $modules ) && is_array( $current_modules ) ) {
-			$new_active_modules = array_diff( $modules, $current_modules );
-			foreach( $new_active_modules as $module ) {
-				/**
-				 * Fires when a specific module is activated.
-				 *
-				 * @since 1.9.0
-				 *
-				 * @param string $module Module slug.
-				 * @param boolean $success whether the module was activated. @since 4.2
-				 */
-				do_action( 'jetpack_activate_module', $module, $success );
-
-				/**
-				 * Fires when a module is activated.
-				 * The dynamic part of the filter, $module, is the module slug.
-				 *
-				 * @since 1.9.0
-				 *
-				 * @param string $module Module slug.
-				 */
-				do_action( "jetpack_activate_module_$module", $module );
-			}
+	static function isArray( $arr ) {
+		if ( ! is_array( $arr ) ) {
+			return array();
+		}
+		return $arr;
+	}
 
-			$new_deactive_modules = array_diff( $current_modules, $modules );
-			foreach( $new_deactive_modules as $module ) {
-				/**
-				 * Fired after a module has been deactivated.
-				 *
-				 * @since 4.2.0
-				 *
-				 * @param string $module Module slug.
-				 * @param boolean $success whether the module was deactivated.
-				 */
-				do_action( 'jetpack_deactivate_module', $module, $success );
-				/**
-				 * Fires when a module is deactivated.
-				 * The dynamic part of the filter, $module, is the module slug.
-				 *
-				 * @since 1.9.0
-				 *
-				 * @param string $module Module slug.
-				 */
-				do_action( "jetpack_deactivate_module_$module", $module );
-			}
+	static function update_active_modules( $modules ) {
+		$current_modules      = Jetpack_Options::get_option( 'active_modules', array() );
+		$active_modules       = Jetpack::get_active_modules();
+	
+		$new_active_modules   = array();
+		$new_deactive_modules = array();
+		$current_modules      = ( is_array( $current_modules ) ) ? $current_modules : array();
+		$modules              = ( is_array( $modules ) ) ? $modules : array();
+		$active_modules       = ( is_array( $active_modules ) ) ? $active_modules : array();
+
+		$new_active_modules   = array_diff( $modules, $current_modules );
+		$new_deactive_modules = array_diff( $active_modules, $modules );
+		$new_current_modules  = array_diff( array_merge( $current_modules, $new_active_modules ), $new_deactive_modules );
+		$success              = Jetpack_Options::update_option( 'active_modules', array_unique( $new_current_modules ) );
+		
+		foreach( $new_active_modules as $module ) {
```

Comment:
Since we're touching these lines anyway, it would be great if we could fix the linting errors in the meantime.

```suggestion
		foreach ( $new_active_modules as $module ) {
```

Reply:
Done in [e93c58d](https://github.com/Automattic/jetpack/pull/12270/commits/e93c58dd7aae4af394b22b7ee0fe6a7afc4469e6).